Split all overlayed funcs, ID's several structs and fields, ID'd game's "scenes"

blackgamma7 opened the pull request:
plus a few months misc work of ID'ing and matching funcs,

blackgamma7 committed:
Added OS Symbols, Alphabet Define table

Added OS symbols from scan and #define of game's charset.

blackgamma7 committed:
ID'd and organized some OS scripts, ID'd OSContPads

ASM will need spring cleaning when splat issue is fixed.

blackgamma7 committed:
Symbol Additions and cleanup

Added some more global symbols and cleaned up some of the definitions of existing ones for a more unified naming convention.

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
labeled SFX functions

There several wrappers for the Sound effect func (several unused) that have been id'd and some already matched. Also cleaned a little of the code from some old warnings.

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
A few matching funcs, a few Id'd data, and headers with SFX definitions

got a few matches and ID's some funcs. Also created header files with info on the charset and some SFX ID's.

blackgamma7 committed:
Added Save game and Sound symbols, updated OS file split

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
one more typo fix

blackgamma7 committed:
id'd load save file fumctions

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
labeles all OS splits, noted splits needed for boot.c

boot.c is actually 3 files

blackgamma7 committed:
rodata_#### made its way back in. oops.

blackgamma7 committed:
updated asm's to match removed rodata

Will get to work re-labeling them.

blackgamma7 committed:
Started Work on GameSave

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
MERGE after history clear.

blackgamma7 committed:
Music and sprite are sepeate files.

Haunted by splat bug. Beware.

blackgamma7 committed:
fixed typo, added a couple matches

blackgamma7 committed:
REMOVE TYPEDEF

blackgamma7 committed:
CLANG'd

blackgamma7 committed:
A few more matches

TODO: clean up for PR, got NameEntry data symbols to work.

blackgamma7 committed:
a few more matches.

blackgamma7 committed:
progress update

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
Aplhabet additions: thinner uses unique data

blackgamma7 committed:
added notes to data

accidentally deleted an #endif. fixed that.

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
personal branch

blackgamma7 committed:
A few easy matches

blackgamma7 committed:
HP is signed.

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master' into bg7

blackgamma7 committed:
Adapt merge, more matches, remove wrong structs

blackgamma7 committed:
More matches

up to 2.2%

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master' into bg7

blackgamma7 committed:
some music.c matches

up to 2.4% now.

blackgamma7 committed:
more files, more matches

blackgamma7 committed:
3% matching

blackgamma7 committed:
3.09%

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master' into bg7

blackgamma7 committed:
Merge branch 'bg7'

blackgamma7 committed:
merge from personal branch, added get/set Yellow gem match

>PI progress in matching
TODO: cleanup style to Standards of upstream.

blackgamma7 committed:
Added more data Definitions

added definitions for sound structs and an array of ascending and descending float values.

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
Updated Symbol Table

blackgamma7 committed:
Revert "Added more data Definitions"

This reverts commit e51f816ea16e3b2d85cf455b045ae730d7803dc9.

blackgamma7 committed:
prepare for merge with upstream

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
3.76 percent match

almost 6% decompiled

blackgamma7 committed:
one more match, declared screen position unions.

blackgamma7 committed:
New union, unmated code.

blackgamma7 committed:
ID'd unused data, better explanation on name entry languages

blackgamma7 committed:
REMOVE TYPEDEF

also gave non-matchings proper encapsulation

blackgamma7 committed:
CLANG'd

also almost 7% decompiled.

blackgamma7 committed:
updated OS and music.c symbols

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
some typedefs and a clairification.

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
4.12 percent matching

also solved a few structs using Mtx's

blackgamma7 committed:
minor update

blackgamma7 committed:
Merge remote-tracking branch 'upstream/master'

blackgamma7 committed:
couple more matches, some helpful comments

blackgamma7 committed:
More matches, more symbols, more splits.

blackgamma7 committed:
4.5% matching, including text funcs

also, lookup table was for cosine,

blackgamma7 committed:
many steps forward, marked as steps back

splitting DMA'd funcs counts as more code bytes in progress.py. also made common.h hold more headers to make #include less cluttered,

blackgamma7 committed:
all overlay funcs split

TODO: have Ghidra tool discern overlay symbols (omit for now?)

blackgamma7 committed:
Mateched "Carberus Alpha" and "final Battle" a little

blackgamma7 committed:
makefile test

Issue with linker - new dirs not recognized

blackgamma7 committed:
Merge branch 'bg7'

blackgamma7 committed:
makefile Error

blackgamma7 committed:
Fixed makefile issue, expanded on symbol table

NOTE: Ghidra script or project needs refining: some func sizes had to manually adjusted.

blackgamma7 committed:
merge with upstream

Drahsid commented:
Thanks for the PR, I am reviewing it now

Drahsid commented on include/Alphabet.h:
```diff
@@ -59,62 +59,63 @@
+#define ALPHA_THIN_Cap_W 0x0071
+#define ALPHA_THIN_Cap_X 0x0072
+#define ALPHA_THIN_Cap_Y 0x0073
+#define ALPHA_THIN_Cap_Z 0x0074
```
Keep all-capitals for global defines

Drahsid commented on include/Scene.h:
```diff
@@ -0,0 +1,75 @@
+#define SCENE_INNERSTRUGGLE	    0x4f
+#define SCENE_UNK80	            0x50 //crash. data suggest more stages were planned for world 5.
+#define SCENE_MERCO	            0x55
+#define SCENE_UNK86	            0x56 //loads "trapped"'s map with marina below. uses "Merco"'s camera effect
```
mixed tabs and spaces? convert to spaces.

Drahsid commented on include/actor.h:
```diff
@@ -103,7 +103,7 @@ typedef void (*ActorFunc)(uint16_t index);
 
 typedef struct {
-    /* 0x000 */ uint8_t unk_0x00[0x80];
+    /* 0x000 */ Mtx Mtxs[2]; //one for each FB
```
rename to be camelCase. I should make this clear for struct variables in the style guide.

Drahsid commented on include/actor.h:
```diff
@@ -124,13 +124,13 @@ typedef struct {
+    /* 0x0B8 */ float ScaleY;
+    /* 0x0BC */ float RotateX;
+    /* 0x0C0 */ float RotateY; 
+    /* 0x0C4 */ float RotateZ;
```
rename to be camelCase. Remove trailing whitespace from rotateY

Drahsid commented on include/actor.h:
```diff
@@ -234,9 +245,17 @@ typedef struct {
+            };
+        };
+    /* 0x190 */ void* unk_0x190;
+    /* 0x194 */ uint8_t unk_0x194[4]; //unused field?
```
On the note of the comment here, due to the Mtx at the starts, it could be due to 8-byte alignment.

Drahsid commented on include/actor.h:
```diff
@@ -250,14 +269,29 @@ typedef struct {
+#define ACTORTYPE_CROSSHAIR 0X71
+#define ACTORTYPE_CLANPOT 0X79
+#define ACTORTYPE_MSHINT 0X7A
+#define ACTORTYPE_REDGEMRING 0X7B
```
Turn this into an enum, change the 0X to 0x for consistency.

Drahsid commented on include/structs.h:
```diff
@@ -0,0 +1,170 @@
+#include "inttypes.h"
+#include <ultra64.h>
+
+enum {
+    GAMESTATE_SOFTRESET,
+    GAMESTATE_INTRO,
+    GAMESTATE_TITLESCREEN,
+    GAMESTATE_DEBUG_SOUNDTEST,
+    GAMESTATE_DEBUG_STAGESELECT,
+    GAMESTATE_LOADING,
+    GAMESTATE_GAMEPLAY,
+    GAMESTATE_CONTINUE, // game over
+    GAMESTATE_UNKNOWN0,
+    GAMESTATE_UNKNOWN1,
+    GAMESTATE_ATTRACT,
+    GAMESTATE_FILESELECT,
+    GAMESTATE_TRANSITION,
+    GAMESTATE_NONE,
+    GAMESTATE_UNKNOWN2 // level select (best times?)
+};
```
If this file is for structs, this should be in common.h instead

Drahsid commented on include/structs.h:
```diff
@@ -0,0 +1,170 @@
+#ifndef STRUCTS_H
+#define STRUCTS_H
```
consider renaming this file to `common_structs.h` for more clarity

Drahsid commented on include/structs.h:
```diff
@@ -0,0 +1,170 @@
+    /*0x04*/ uint16_t ActorFeild0xD8;
+    /*0x06*/ uint16_t unk_0x6;
+    /*0x08*/ uint16_t ActorType;
+}ClanpotItem;
```
use camelCase for struct variable names

Drahsid replied:
Additionally, for TextTransform, add offsets and sizeof
Additionally, for ClanpotItem, space offsets and add sizeof

Drahsid commented on mischiefmakers.map:
```diff
@@ -22,30 +22,28 @@ Discarded input sections
  .bss           0x0000000000000000      0x200 build/src/music.c.o
```
We might want to gitignore this file

Drahsid commented on mischiefmakers.yaml:
```diff
@@ -14,57 +14,63 @@ segments:
     vram: 0x80000400
     subsegments:
-      - [0x01000, asm, entrypoint]
+      - [0x01000, hasm, entrypoint]
```
consider moving hasm into `../src/...` so that we can comment on it, and so that it doesn't get resplit every time the repo is nuked.

Drahsid commented on mischiefmakers.yaml:
```diff
@@ -238,35 +249,649 @@ segments:
+  - [0x4A1FBC,bin] 
+  #more romcopied bits. looks to be compressed data
+  - [0x4A1FC0,bin]
+#call lists and code segments loaded per scene by RomCopy_Funcs
```
Space all of this consistently. Tab each entry forward by 2 spaces, add spaces before the commas, and before the comment text

Drahsid commented on mischiefmakers.yaml:
```diff
@@ -238,35 +249,649 @@ segments:
+    overlay: true
+    subsegments:
+      - [0x713630, c,ActorFuncs1/General]
+      - [0x719150,bin,ActorFuncs1/General]
```
Space the commas in the subsegment entries

Drahsid commented on mischiefmakers.yaml:
```diff
@@ -238,35 +249,649 @@ segments:
+    subsegments:
+    - [0x7EA730, c,ActorFuncs5/World4]
+    - [0x7EADE0, bin,ActorFuncs5/World4]
+    #- [0x7EAFA0, rodata,ActorFuncs5/World4]
```
Space the commas in the subsegment entries

Drahsid commented on include/data_symbols.h:
```diff
@@ -221,56 +194,97 @@ extern uint16_t gNameEntrySpace[11];
 extern uint16_t gNameEntryRow4ENG[18];
 extern UNK_TYPE D_800C71A0;
-extern UNK_TYPE D_800C7E14;
+extern u16 D_800C7E14[16]; //"G......AME......OV...E...R"
```
use `uint16_t`

Drahsid commented on include/data_symbols.h:
```diff
@@ -221,56 +194,97 @@ extern uint16_t gNameEntrySpace[11];
 extern UNK_TYPE D_800C71A0;
-extern UNK_TYPE D_800C7E14;
+extern u16 D_800C7E14[16]; //"G......AME......OV...E...R"
+extern char*D_800C81E4[22]; //used by stage select in building menu.
```
spacing, use uint8_t* unless this is text

Drahsid commented on include/data_symbols.h:
```diff
@@ -221,56 +194,97 @@ extern uint16_t gNameEntrySpace[11];
-extern uint8_t D_800CCFDC[88]; // Stage BGM indices
-extern int32_t D_800CEC0C;
-extern uint8_t D_800D16AA[]; // right in the middle of a pointer array?
+extern void (*Scene_CameraInits[88])(void); //each scene's CameraInit function
+extern void (*Scene_CameraFuncs[88])(void); //each scene's CameraTick function
```
Consider creating a typedef for this

Drahsid commented on include/function_symbols.h:
```diff
@@ -241,167 +235,168 @@ extern void func_8001FF30(void);
+extern int16_t Text_Print3Digits();
+extern uint16_t Text_GetWidth(uint16_t* txt);
+extern uint16_t Text_getKerning(uint16_t* txt);
+extern uint16_t Text_ZeroFlagActors(uint16_t index, uint16_t* TXT);
+extern uint16_t Text_PrintAlphaAt(uint16_t index, uint16_t* TXT, uint16_t pos_x, uint16_t pos_y,uint16_t pos_z);
+extern uint16_t Text_PrintAlphaAtColor(uint16_t index,uint16_t *TXT,uint16_t pos_x,uint16_t pos_y,uint16_t pos_z,uint8_t r,uint8_t g,uint8_t b);
+extern uint16_t Text_PrintAlphaAtColorScale(int16_t, uint16_t*, int32_t, int32_t, uint16_t, uint8_t, uint8_t, uint8_t, float, float);
+extern uint16_t Text_PrintAlphaAt2(uint16_t index, uint16_t* TXT, uint16_t pos_x, uint16_t pos_y,uint16_t pos_z);
+extern uint16_t Text_PrintAlphaAt3(uint16_t index, uint16_t* TXT, uint16_t pos_x, uint16_t pos_y,uint16_t pos_z);
```
consider making local variable TXT conform to snake_case, space commas

Drahsid commented on include/function_symbols.h:
```diff
@@ -428,25 +423,25 @@ extern void func_80028E1C();
-extern void func_80029798();
-extern int32_t func_8002981C(int32_t A, int32_t B, int32_t C);
-extern float func_80029860(float A, float B, float C);
+extern int32_t func_800294E0(int32_t A, int32_t B);
+extern uint32_t func_800295D8(int32_t A, int32_t B);
+extern uint32_t func_80029600(int32_t A, int32_t B);
+extern uint32_t func_80029628(int32_t A, int32_t B);
+extern uint32_t func_80029650(int32_t A, int32_t B);
```
A and B should be snake_case

Drahsid commented on include/function_symbols.h:
```diff
@@ -1522,15 +1517,15 @@ extern void func_8008379C(int16_t x, int16_t y, int16_t txt, int16_t param_4);
 extern void DebugText_PrintShortHexBlack(int16_t s, int posX, int posY);
 extern uint64_t DebugText_GetTime(void);
-extern void DebugText_PrintDeltaTime();
+extern void DebugText_PrintDeltaTime(int posX, int posY);
```
arguments should be snake_case

Drahsid commented on src/134E0.c:
```diff
@@ -1,23 +1,22 @@
+uint8_t func_80012944(int32_t x, int32_t y) {
+    if((y&0xf)<(x&0xf)) return 255;
+    return 0;
+}*/
```
make sure each function has a newline between the next, and change this to NON_MATCHING. This should work on latest splat, if you're having troubles let me know and we can fix it

Drahsid commented on src/156F0.c:
```diff
@@ -1,23 +1,32 @@
+/*
+void func_800151D8(uint16_t index){
+    func_80014FD0(index,((gActors[index].pos.x+gScreenPosCurrentX._hi&0xf)>>1) +8);
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/156F0.c:
```diff
@@ -53,17 +62,33 @@ void func_80016CB4(void) {
+                gActors[index].flag=0;
+        }
+    }
+}
```
consider spacing out the operators here for better readability

Drahsid commented on src/17A70.c:
```diff
@@ -213,20 +209,39 @@ void func_8001751C(void) {
+    }
+    if((gButtonPress&button==0) && (*sel!=20)) return 0;
+    return 1;
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/17A70.c:
```diff
@@ -406,24 +421,25 @@ void Intro_Tick(void) {
-    func_800276DC(0x60U, &D_800C8F88, 0xFFA6, 0xFFAE, 0, func_80027588(2U, 0x1FU, 0x1FU, 0x18U));
+    Text_PrintASCII(0x39U, &D_800C8F68, 0xFFCA, 0xFFE4, 0, Text_ConvertColor(0U, (0x1F - D_801781A0[0] / 4), (0x1F - D_801781A0[0] / 4), 0x1FU));
+    Text_PrintASCII(0x49U, &D_800C8F74, 0xFFAA, 0xFFC0, 0, Text_ConvertColor(2U, 0x1FU, 0x1FU, 0x18U));
+    Text_PrintASCII(0x60U, &D_800C8F88, 0xFFA6, 0xFFAE, 0, Text_ConvertColor(2U, 0x1FU, 0x1FU, 0x18U));
```
if the Us next to the hex here aren't required for the OK, remove them for readability.

Drahsid commented on src/17A70.c:
```diff
@@ -1077,7 +1094,7 @@ void func_8001C834(void) {
 #pragma GLOBAL_ASM("asm/nonmatchings/17A70/func_8001C8B0.s")
 
-#pragma GLOBAL_ASM("asm/nonmatchings/17A70/func_8001C97C.s")
+#pragma GLOBAL_ASM("asm/nonmatchings/17A70/printIcon_YellowGem.s")
```
use Snake_PascalCase for functions

Drahsid commented on src/1F1E0.c:
```diff
@@ -225,7 +222,7 @@ void YellowGem_PrintProgress(void) {
 
 #ifdef NON_MATCHING
-void func_80020844(void) { // resets sound levels after exiting pause menu?
+void PauseGame_RestoreVolume(void) { // resets sound levels after exiting pause menu?
```
I think `Gameplay_RestoreVolume` makes more sense given the context

Drahsid commented on src/1F1E0.c:
```diff
@@ -234,17 +231,17 @@ void func_80020844(void) { // resets sound levels after exiting pause menu?
 #endif
 
-void func_800208D4(void) {
+void PauseGame_Unpause(void) {
```
I think `Gameplay_Unpause` makes more sense given the context

Drahsid commented on src/22290.c:
```diff
@@ -1,24 +1,29 @@
+}*/
 
-#pragma GLOBAL_ASM("asm/nonmatchings/22290/func_800218FC.s")
+#pragma GLOBAL_ASM("asm/nonmatchings/22290/Gfx_DrawRectange.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/22290.c:
```diff
@@ -71,77 +76,89 @@ void func_80022F48(void) {
+    for(i=0;i<66;i++){
+        D_80104098[i].Active=0;
+    }
+}
```
Consider spacing operators for better readability

Drahsid commented on src/438E0.c:
```diff
@@ -430,14 +428,14 @@ void func_80046A30(void) {
+#pragma GLOBAL_ASM("asm/nonmatchings/438E0/transition_expandingSqares.s")
 
-#pragma GLOBAL_ASM("asm/nonmatchings/438E0/func_80046B4C.s")
+#pragma GLOBAL_ASM("asm/nonmatchings/438E0/Transition_fadeIn.s")
```
Use Pascal_SnakeCase for function names

Drahsid commented on src/48A30.c:
```diff
@@ -0,0 +1,38 @@
+void func_800483B4(uint16_t index){
+    if(gActors[index].unk_0xD1==3)gActors[index].unk_0x150._w=600;
+    D_800BE5F4=1;
+}
```
Consider consistently spacing operators for better readability. Make sure functions have a newline between them as well.

Drahsid commented on src/241E0.c:
```diff
@@ -1,93 +1,101 @@
+}*/
 
-#pragma GLOBAL_ASM("asm/nonmatchings/241E0/func_800235F4.s")
+#pragma GLOBAL_ASM("asm/nonmatchings/241E0/Camera_ApplyRotate.s")
```
use NON_MATCHING ifdef gate

Drahsid commented on src/241E0.c:
```diff
@@ -155,51 +163,69 @@ void func_80024074(void) {
 
-#pragma GLOBAL_ASM("asm/nonmatchings/241E0/func_80024528.s")
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/241E0/CameraTick_Vertigo.s")
```
use NON_MATCHING ifdef gate

Drahsid commented on src/241E0.c:
```diff
@@ -272,42 +301,51 @@ void func_8002515C(void) {
+            gScreenPosTargetY._hi-=0x400;
+        }
+    }
+}
```
Consider spacing operators for better readability

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    func_80027644(index+1,N*2+0x2D2,pos_x+9,pos_y,pos_z,arg5);
+    return index+2;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/Text_Print2Digits.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    return Text_GetWidth(TXT)+Text_GetWidth(TXT+1);
+}*/
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/Text_getKerning.s")
```
Use Pascal_SnakeCase for function names. Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+        pos_x+=16;
+    }
+    return index;
+}
```
Consider spacing operators for better readability

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    D_800CA230=0;
+    if(gPlayerActor.health<0) gPlayerActor.health=0;
+
+}
```
Consider spacing operators for better readability

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+        gActors[index].vel.y_w=0;
+    }
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800299B0.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    if(y<0)y++;
+    return ((x>>1)+gActors[B].pos.y+(y>>1))-gActors[A].pos.y;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_8002A990.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+        p0+=2;
+        i=*p0;
+    }
+}
```
Use uint16_t. Consider spacing operators for better readability

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    if(gActors[index].unk_0x110==0.0)gActors[index].flag=0;
+    else gActors[index].unk_0x110-=1.0;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/ActorTick_1.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    gActors[index].vel.y_w+=gActors[index].unk_0x15C;
+    gActors[index].unk_0x150._w = func_8002B5A0(gActors[index].unk_0x150._lo,gActors[index].unk_0x168._lo,gActors[index].unk_0x164._lo,gActors[index].unk_0x160._lo);
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80033DE4.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+        gActors[index].flag=0;
+    }
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_800348E4.s")
```
immediate float rvalues are considered rodata as far as I know. Make sure argument types are correct, and make sure you have the rodata section correctly defined. Use NON_MATCHING ifdef gate. Consider spacing operators for better readability

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    gActors[i].ScaleY=gActors[i].unk_0x16C /1000.0f;
+}
+
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/clanpot_moveitems.s")
```
Use Pascal_SnakeCase for function names

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    gActors[i].unk_0xB0=-10;
+    gActors[i].unk_0xAA=-8;
+    gActors[i].unk_0xAC=8;
+}
```
Consider spacing operators for better readability

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    if(gActors[index].unk_0xD0_h==0) func_800358DC(index);
+    else func_80035A20(index);
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_8003667C.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    gActors[index].unk_0x98 |=0x2000;
+    Actor_Grayscale(index,127);
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_80039894.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    func_8005CA34(8,60);
+    return 0xC0;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/func_8003D628.s")
```
Use NON_MATCHING ifdef gates

Drahsid commented on src/27F70.c:
```diff
@@ -0,0 +1,1743 @@
+    gActors[index].pos.y_w = gActors[parent].pos.y_w;
+    gActors[index].pos.z_w = gActors[parent].pos.z_w + 0x200000;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/27F70/Crosshair_CopyCoords.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/4FEB0.c:
```diff
@@ -0,0 +1,171 @@
+     return 2;
+ }
+ return 1;
+}
```
Tabs should be 4 spaces. Consider spacing operators for better readability.

Drahsid commented on src/4FEB0.c:
```diff
@@ -0,0 +1,171 @@
+    gActors[index].unk_0xD0_h=42;
+    gActors[index].unk_0xF8=gActors[index].vel.x_w;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/4FEB0/func_80058F54.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/5D120.c:
```diff
@@ -0,0 +1,69 @@
+    if(x&0x30==0x30)return x & 0xC0;
+    return x;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/5D120/func_8005C870.s")
```
Use NON_MATCHING ifdef gates

Drahsid commented on src/5D120.c:
```diff
@@ -0,0 +1,69 @@
+    if(x!=b) gActors[index].unk_0x170_w = a;
+    return a!=b;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/5D120/func_8005D370.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/5E230.c:
```diff
@@ -0,0 +1,118 @@
+void func_8005E09C(uint32_t i,int16_t x){
+    D_801783F8[i]=x;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/5E230/func_8005E09C.s")
```
Use NON_MATCHING ifdef gates

Drahsid commented on src/61B80.c:
```diff
@@ -0,0 +1,124 @@
+        gActors[index].rgba.a = gActors[ACTOR_COUNT0].rgba.a;
+    }
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/61B80/func_80061350.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/66250.c:
```diff
@@ -74,23 +71,31 @@
-#pragma GLOBAL_ASM("asm/nonmatchings/66250/func_80068888.s")
 
-#pragma GLOBAL_ASM("asm/nonmatchings/66250/func_80068898.s")
+uint32_t func_80068888(uint32_t x, uint32_t y){return 0;}
```
Make these multi-line

Drahsid commented on src/66250.c:
```diff
@@ -102,35 +107,96 @@
 #pragma GLOBAL_ASM("asm/nonmatchings/66250/func_80068A68.s")
 
-#pragma GLOBAL_ASM("asm/nonmatchings/66250/func_80068ACC.s")
+uint32_t func_80068ACC(){return 1;}
```
Make this multi-line

Drahsid commented on src/66250.c:
```diff
@@ -102,35 +107,96 @@
+       return 1;
+    }
+    return 0;
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/66250.c:
```diff
@@ -102,35 +107,96 @@
+       return 1;
+    }
+    return 0;
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/66250.c:
```diff
@@ -293,7 +359,10 @@ void func_8006C5A4(uint16_t index) {
+/*
+uint16_t func_8006C8B8(uint16_t x){
+    return func_80069C54(x) | func_8006C7B8(x) | func_80069884(x);
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/66250.c:
```diff
@@ -305,11 +374,17 @@ void func_8006C5A4(uint16_t index) {
-
+/*uint16_t func_8006CB4C(uint16_t index){
+    return func_8006C7B8(index)|func_80069884(index);
+}*/
```
Use NON_MATCHING ifdef gates

Drahsid commented on src/66250.c:
```diff
@@ -323,7 +398,14 @@ void func_8006C5A4(uint16_t index) {
+        func_8006CD5C(index);
+        func_800358DC(index);
+    }
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/66250.c:
```diff
@@ -454,14 +539,24 @@ void func_8006C5A4(uint16_t index) {
+            gActors[index].RotateZ=0.0f;
+    }
+    else {func_80072E54(index);}
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/66250.c:
```diff
@@ -608,7 +708,16 @@ void func_8007325C(uint16_t index) {
+        gActors[index].unk_0x150._w |=2;
+    }
+    else{gActors[index].unk_0x150._w &= ~2;}
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/7D8E0.c:
```diff
@@ -0,0 +1,90 @@
+    gActors[index].unk_0x15C=q;
+    gActors[index].unk_0x14C=j;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/7D8E0/func_8007CD68.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/7D8E0.c:
```diff
@@ -0,0 +1,90 @@
+    }
+    return g;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/7D8E0/func_8007D1E8.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/82920.c:
```diff
@@ -1,13 +1,25 @@
+/*
+void func_80082184(uint16_t i){
+    func_80081E38(i);
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/84BB0.c:
```diff
@@ -0,0 +1,244 @@
+    gActors[index].unk_0x174=0;
+    gActors[index].unk_0x178=0;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/84BB0/func_80085F08.s")
```
Use NON_MATCHING ifdef gates. Consider putting `extern` variables in the relevant headers.

Drahsid commented on src/8D0A0.c:
```diff
@@ -16,22 +12,39 @@ void func_8008C4E0(uint16_t index) {
+    gActors[16].flag=0;
+    gActors[16].unk_0xD0_h=0;
+    
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/8D0A0.c:
```diff
@@ -77,26 +96,66 @@ void StartContinueMode(uint16_t index) {
-
-#pragma GLOBAL_ASM("asm/nonmatchings/8D0A0/func_8008E3C0.s")
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/8D0A0/ActorTick_MarinaOhNo.s")
```
Use NON_MATCHING ifdef gates

Drahsid commented on src/8F080.c:
```diff
@@ -111,12 +114,17 @@
+    uint32_t ret =func_80092FD4(x);
+    if(ret) ret = func_8005DFC8(0);
+    return ret;
+}*/
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/8F080.c:
```diff
@@ -156,11 +164,17 @@
+        gActors[index].unk_0x118-=360;
+        SFX_Play_2(0x116); //dash sound
+        }
+}
```
Cleanup tabbing. Consider spacing operators for better readability.

Drahsid commented on src/A540.c:
```diff
@@ -10,26 +10,26 @@
 }
 
-void func_800109B0(void) {
+void lookAt_Reset(void) {
```
Use Pascal_SnakeCase for function names

Drahsid commented on src/ActorFuncs1/CerberusAlpha.c:
```diff
@@ -0,0 +1,478 @@
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs1/CerberusAlpha/func_801922A0_68A2A0.s")
+void func_8019231C_68A31C(uint16_t x){
+    gActors[193].flag=0;
+}
```
Add newlines between functions

Drahsid commented on src/ActorFuncs1/CerberusAlpha.c:
```diff
@@ -0,0 +1,478 @@
+
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs1/CerberusAlpha/func_801A3858_69B858.s")
+
+void func_801A3D1C_69BD1C(uint16_t x){}
```
One tab should be 4 spaces

Drahsid commented on src/ActorFuncs1/Ending.c:
```diff
@@ -0,0 +1,552 @@
+    gActors[index].rgba.a=a;
+    gActors[index].unk_0x154._w= (uint32_t)a<<16;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs1/Ending/func_80192634_6F98C4.s")
```
use NON_MATCHING ifdef gate

Drahsid commented on src/ActorFuncs1/FinalBattle.c:
```diff
@@ -0,0 +1,433 @@
+    if(gActors[index].unk_0x138!=0.0) x=index+15;
+    gActors[x].unk_0x84=0x203A;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs1/FinalBattle/func_8019232C_6BCFFC.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/ActorFuncs1/SasquatchBeta.c:
```diff
@@ -0,0 +1,137 @@
+        gActors[79].flag=2;
+    }
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs1/SasquatchBeta/func_80192340_6A0340.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/ActorFuncs2/Festival.c:
```diff
@@ -0,0 +1,30 @@
+    DebugText_PrintShortHexWhite(D_800E3584>>0x10,gActors[index].pos.x,0);
+    if(D_800E3584&0xc0000) gActors[index].unk_0xD0_h=16;
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs2/Festival/func_8019B6A8_753E28.s")
```
Use NON_MATCHING ifdef gates

Drahsid commented on src/ActorFuncs2/Flambee.c:
```diff
@@ -0,0 +1,144 @@
+    f32 f3 = (f32) i;
+    if (f3 == 0.0f) return f2;
+    return ((f2 - f1) / f3) + f1;
+}
```
Use float and int32_t

Drahsid commented on src/ActorFuncs3/World4.c:
```diff
@@ -0,0 +1,352 @@
+    }
+}*/
+
+#pragma GLOBAL_ASM("asm/nonmatchings/ActorFuncs3/World4/func_801A6908_7754F8.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/romCopy.c:
```diff
@@ -0,0 +1,80 @@
+    RomCopy_A(0x1E6900,&D_80267CD0,0X2E240);
+    D_801376E0=&D_80200400+func_80004910(&D_80267CD0,&D_80200400);
+}*/
+#pragma GLOBAL_ASM("asm/nonmatchings/RomCopy/func_80025E00.s")
```
Use NON_MATCHING ifdef gate

Drahsid commented on src/romCopy.c:
```diff
@@ -0,0 +1,80 @@
+
+#ifdef NON_MATCHING
+// the subtraction gets optimized out...
+s32 func_8002653C(void) {
```
Use int32_t

Drahsid commented on tools/7754F0_old.c:
```diff
@@ -1,6 +1,6 @@
 #include "actor.h"
-#include "data_symbols.h"
-#include "function_symbols.h"
+
```
Was this a mistake? You can ignore files locally by adding them to the exclude file

Drahsid reviewed (changes requested):
Good work! Most issues related to using comments over ifdef gates. Basically do the cleanup that I described in my review messages and this will be merged.

blackgamma7 committed:
little cleanup, lotta matches

found some overlayed scripts and alphabet indicies i missed late time as well.

blackgamma7 committed:
a few matches, clairfiing a field and some helpful macros.

blackgamma7 committed:
new structs id, new macros, ID'd fields, more matches

blackgamma7 committed:
back up to 3% matching

also added #defines for the flags used by the gem spawner

blackgamma7 committed:
couple more matches

Also added the bss for "cerberus Alpha"

blackgamma7 committed:
added notes and #defines for overlay funcs

blackgamma7 committed:
couple more matches, split some tables, added compile flags for os

blackgamma7 committed:
forgot a symbol update

blackgamma7 committed:
ID'd actor fields, more matches

also added more detail about "migen Brawl"

blackgamma7 committed:
added code from Japan-only text bubbles.

blackgamma7 committed:
revised Actor struct with new union, split and renamed a few scrips, started adding Japanese Alphabet indecies

Created a union with every common 32-bit datatype and used that in the actor struct (suspect Treasure did the same.)

blackgamma7 committed:
unified "thisActor" #define, added float tables, a few more matches

Put the cosine and unused float tables in c scripts, replaced with "thisActor" where applicable, changed some file name, and added a couple matches

blackgamma7 committed:
More file splits, few more matches

attempted to combine libultra rodata back into files, no dice. Splat problem?

blackgamma7 committed:
Added some non-matchings, defined some ending data,  matched some easy stuff

Also discerned more code from Japanese version, see text.c.

blackgamma7 committed:
updated actor fields, renamed player manager struct, a handful of matches

ID'd a few more actor fields RE: position and speed. also found some more easy matches.
